src/editor/ui_menus/field_info_window.cc

```diff
@@ -0,0 +1,257 @@
+		/** TRANSLATORS: Editor terrain property: port space */
+		caps_strings.push_back(_("port"));
+	}
+	if (caps & Widelands::MOVECAPS_WALK) {
+		/** TRANSLATORS: Editor terrain property: units can walk on this terrain */
+		caps_strings.push_back(_("walkable"));
+	}
+	if (caps & Widelands::MOVECAPS_SWIM) {
+		/** TRANSLATORS: Editor terrain property: units can swim on this terrain (fish, ships) */
+		caps_strings.push_back(_("swimmable"));
+	}
+
+	buf += as_listitem(
+	   bformat(_("Caps: %s"), i18n::localize_list(caps_strings, i18n::ConcatenateWith::COMMA)),
+	   font_style);
+}
+
+void FieldInfoWindow::add_owner_info(std::string& buf) {
+	if (f_.get_owned_by() > 0) {
+		buf += as_listitem(
+		   bformat(_("Owned by: Player %u"), static_cast<unsigned int>(f_.get_owned_by())),
+		   font_style);
+	} else {
+		buf += as_listitem(_("Owned by: —"), font_style);
+	}
+}
+
+void FieldInfoWindow::add_terrain_info(std::string& buf) {
+	buf += as_heading(_("Terrain"), UI::PanelStyle::kWui);
+
+	const Widelands::TerrainDescription* ter = parent_.egbase().descriptions().get_terrain_descr(
+	   center_.triangle.t == Widelands::TriangleIndex::D ? tf_.terrain_d() : tf_.terrain_r());
+
+	buf += as_listitem(bformat(pgettext("terrain_name", "Name: %s"), ter->descname()), font_style);
+
+	std::vector<std::string> terrain_is_strings;
+	for (const Widelands::TerrainDescription::Type& terrain_type : ter->get_types()) {
+		terrain_is_strings.push_back(terrain_type.descname);
+	}
+
+	buf += as_listitem(
+	   bformat(
+	      /** TRANSLATORS: "Is" is a list of terrain properties, e.g. "arable, unreachable and
+	       * unwalkable". You can also translate this as "Category: %s" or "Property: %s" */
+	      _("Is: %s"), i18n::localize_list(terrain_is_strings, i18n::ConcatenateWith::AMPERSAND)),
+	   font_style);
+}
+
+void FieldInfoWindow::add_mapobject_info(std::string& buf) {
+	const Widelands::BaseImmovable* immovable = f_.get_immovable();
+	Widelands::Bob* bob = f_.get_first_bob();
+	if (immovable || bob) {
+		/** TRANSLATORS: Heading for immovables and animals in editor info tool */
+		buf += as_heading(_("Objects"), UI::PanelStyle::kWui);
+		if (immovable) {
+			buf += as_listitem(bformat(_("Immovable: %s"), immovable->descr().descname()), font_style);
+		}
+
+		if (bob) {
+			// Collect bobs
+			std::vector<std::string> critternames;
+			std::vector<std::string> shipnames;
+			std::vector<std::string> workernames;
+			do {
+				switch (bob->descr().type()) {
+				case (Widelands::MapObjectType::CRITTER):
+					critternames.push_back(bob->descr().descname());
+					break;
+				case (Widelands::MapObjectType::SHIP): {
+					if (upcast(Widelands::Ship, ship, bob)) {
+						shipnames.push_back(ship->get_shipname());
+					}
+				} break;
+				case (Widelands::MapObjectType::WORKER):
+				case (Widelands::MapObjectType::CARRIER):
+				case (Widelands::MapObjectType::SOLDIER):
+					workernames.push_back(bob->descr().descname());
+					break;
+				default:
+					break;
```

Comment:
```suggestion
					NEVER_HERE();
```
Shouldn't happen surely?